### ISSUE TITLE
Timezone selection in Maintenance schedule is ignored - we are all living in Africa/Abidjan   now

### ISSUE THREAD

**Issue by uvwild** [gardener/dashboard#944]
Changing the Timezone of the maintenance schedule of a cluster in the gardener webUI  has  no effect. 
Lifecycle
Hibernation
Waking up Cluster...
**Maintenance**
Start time: 01:00 **Africa/Abidjan**

Maybe also it would be meaningful to clarify how the maintenance happens when the cluster is hibernating?!?
Or if that is not working maybe the UI should try to limit the maintenance schedule to the wake times?

**Comment by grolu** [comment]
Thank you @rfranzke I will have a look

**Comment by grolu** [comment]
Hi @uvwild. We use momentjs to detect your current location. This information is used to display your current location in the Dashboard and we calculate the maintenance time to match your current location. My guess is that something went wrong while determining your location. What kind of browser do you use and did you apply additional privacy settings?

Nevertheless, we need to improve on our side here and handle this case. The best solution is probably to show the maintenance in UTC in case we cannot determine a user's location.

**Comment by uvwild** [comment]
OK, but why you offer a drop down menu for the timezone which is then
ignored?

Also the Africa tz is NOT in my browser settings ;)

However, I also advise strongly against using any hidden browser info for
this!

The browser has nothing to do with cluster maintenance schedules.

This is simply wrong by experienced use cases over the years.

Since there is the GUI element, the obvious solution is to use that value.
It could be preset with some other tz setting of the cluster, but not the
browsers.

Most international systems I worked with used GMT for server tz in my past.

**Comment by grolu** [comment]
Maybe you got this wrong. Of course we do __not__ ignore the value of the dropdown. The Dashboard always stores the maintenance time in UTC. However, we show the maintenance time according to the user's current location. The idea behind this is that an operator based in e.g. PST can immediately see when the maintenance will be scheduled, regardless who created the cluster or adjusted the maintenance time.
In this example, the cluster would show as

`Start time: 02:00 Europe/Berlin` for the cluster owner located in Germany
The operator, in this example located in the pacific west, would see the same maintenance time as `Start time: 5:00pm America/Vancouver`

We did this as we expect operators and users of a (productive) cluster do not necessarily live in the same timezone.

>OK, but why you offer a drop down menu for the timezone which is then ignored?

As I already stated, the maintenance time is stored in UTC by the Dashboard. This means, that we calculate the value based on the selected maintenance time and the timezone offset derived from the location selected in the dropdown.

>Also the Africa tz is NOT in my browser settings ;)

I never suggested this. I asked you if you applied special privacy settings or use a special browser. Africa/Abidjan is the first timezone in the list, so this timezone is currently the fallback if location detection does not work properly. As I already indicated, this is a __bug__. We should fallback to UTC if we cannot determine the user's timezone.

>It could be preset with some other tz setting of the cluster, but not the browsers

The cluster itself has no timezone. It would be possible to not store the time in UTC but use the offset to indicate in which timezone the cluster maintenance time was configured (e.g. GMT+1).

However, we will first fix the issue mentioned above and then we can discuss if we want to change the overall behavior, if this is still required.
